ci/tests/puppeteer/scenarios/mfa-global-multiple-providers/script.js

```diff
@@ -0,0 +1,33 @@
+const puppeteer = require('puppeteer');
+const assert = require('assert');
+
+(async () => {
+    const browser = await puppeteer.launch({
+        ignoreHTTPSErrors: true,
+        headless: true
+    });
+    const page = await browser.newPage();
+    await page.goto("https://localhost:8443/cas/login?service=https://example.org");
+
+    await page.type('#username', "casuser");
+    await page.type('#password', "Mellon");
+    await page.keyboard.press('Enter');
+    await page.waitForNavigation();
+
+    await page.waitForTimeout(9000);
+    const mfaGauth = await page.$('div#mfa-gauth');
+    const form = await mfaGauth.$('form');
+    await form.evaluate(form => form.submit());
+    await page.waitForTimeout(5000);
+
+    let element = await page.$('#login p');
+    await page.type('#token', "123456");
```

Comment:
I've tried to login with one of the scratch code at this point, in order to complete the authentication flow but the following error is thrown:

```
2021-07-23 13:04:25,325 DEBUG [org.apereo.cas.authentication.PolicyBasedAuthenticationManager] - <Authentication credentials provided for this transaction are [[GoogleAuthenticatorTokenCredential(super=OneTimeTokenCredential(token=11040740), accountId=1)]]>
2021-07-23 13:04:25,326 DEBUG [org.apereo.cas.authentication.DefaultAuthenticationEventExecutionPlan] - <Candidate/Registered authentication handlers for this transaction are [[org.apereo.cas.authentication.handler.support.HttpBasedServiceCredentialsAuthenticationHandler@2f4d01b6, org.apereo.cas.adaptors.yubikey.YubiKeyAuthenticationHandler@53cddaf8, org.apereo.cas.authentication.AcceptUsersAuthenticationHandler@652e345, org.apereo.cas.gauth.GoogleAuthenticatorAuthenticationHandler@3214bad]]>
2021-07-23 13:04:25,326 DEBUG [org.apereo.cas.authentication.DefaultAuthenticationEventExecutionPlan] - <Authentication handler resolvers for this transaction are [[org.apereo.cas.authentication.handler.ByCredentialTypeAuthenticationHandlerResolver@1f3cc2c8, org.apereo.cas.authentication.handler.ByCredentialTypeAuthenticationHandlerResolver@739523ab, org.apereo.cas.authentication.handler.RegisteredServiceAuthenticationHandlerResolver@22a63740]]>
2021-07-23 13:04:25,326 DEBUG [org.apereo.cas.authentication.AuthenticationHandlerResolver] - <Default authentication handlers used for this transaction are [HttpBasedServiceCredentialsAuthenticationHandler,YubiKeyAuthenticationHandler,Static Credentials,GoogleAuthenticatorAuthenticationHandler]>
2021-07-23 13:04:25,326 DEBUG [org.apereo.cas.authentication.DefaultAuthenticationEventExecutionPlan] - <Resolved and finalized authentication handlers to carry out this authentication transaction are [[org.apereo.cas.authentication.handler.ByCredentialTypeAuthenticationHandlerResolver@1f3cc2c8, org.apereo.cas.authentication.handler.ByCredentialTypeAuthenticationHandlerResolver@739523ab, org.apereo.cas.authentication.handler.RegisteredServiceAuthenticationHandlerResolver@22a63740]]>
2021-07-23 13:04:25,326 DEBUG [org.apereo.cas.authentication.PolicyBasedAuthenticationManager] - <Candidate resolved authentication handlers for this transaction are [[org.apereo.cas.authentication.handler.support.HttpBasedServiceCredentialsAuthenticationHandler@2f4d01b6, org.apereo.cas.adaptors.yubikey.YubiKeyAuthenticationHandler@53cddaf8, org.apereo.cas.authentication.AcceptUsersAuthenticationHandler@652e345, org.apereo.cas.gauth.GoogleAuthenticatorAuthenticationHandler@3214bad]]>
2021-07-23 13:04:25,326 DEBUG [org.apereo.cas.authentication.PolicyBasedAuthenticationManager] - <Attempting to authenticate credential [GoogleAuthenticatorTokenCredential(super=OneTimeTokenCredential(token=11040740), accountId=1)]>
2021-07-23 13:04:25,326 DEBUG [org.apereo.cas.authentication.PolicyBasedAuthenticationManager] - <Authentication handler [HttpBasedServiceCredentialsAuthenticationHandler] does not support the credential type [GoogleAuthenticatorTokenCredential(super=OneTimeTokenCredential(token=11040740), accountId=1)]. Trying next...>
2021-07-23 13:04:25,327 DEBUG [org.apereo.cas.authentication.PolicyBasedAuthenticationManager] - <Authentication handler [YubiKeyAuthenticationHandler] does not support the credential type [GoogleAuthenticatorTokenCredential(super=OneTimeTokenCredential(token=11040740), accountId=1)]. Trying next...>
2021-07-23 13:04:25,327 DEBUG [org.apereo.cas.authentication.handler.support.AbstractUsernamePasswordAuthenticationHandler] - <Credential is not one of username/password and is not accepted by handler [Static Credentials]>
2021-07-23 13:04:25,328 DEBUG [org.apereo.cas.authentication.PolicyBasedAuthenticationManager] - <Authentication handler [Static Credentials] does not support the credential type [GoogleAuthenticatorTokenCredential(super=OneTimeTokenCredential(token=11040740), accountId=1)]. Trying next...>
2021-07-23 13:04:25,328 DEBUG [org.apereo.cas.authentication.PolicyBasedAuthenticationManager] - <Attempting authentication of [11040740] using [GoogleAuthenticatorAuthenticationHandler]>
2021-07-23 13:04:25,330 DEBUG [org.apereo.cas.otp.repository.token.CachingOneTimeTokenRepository] - <Found used tokens [null]>
2021-07-23 13:04:25,330 DEBUG [org.apereo.cas.gauth.credential.GoogleAuthenticatorOneTimeTokenCredentialValidator] - <Attempting to authorize OTP token [11040740]...>
2021-07-23 13:04:25,333 ERROR [org.apereo.cas.authentication.PolicyBasedAuthenticationManager] - <Authentication has failed. Credentials may be incorrect or CAS cannot find authentication handler that supports [GoogleAuthenticatorTokenCredential(super=OneTimeTokenCredential(token=11040740), accountId=1)] of type [GoogleAuthenticatorTokenCredential]. Examine the configuration to ensure a method of authentication is defined and analyze CAS logs at DEBUG level to trace the authentication event.>
2021-07-23 13:04:25,333 ERROR [org.apereo.cas.authentication.PolicyBasedAuthenticationManager] - <[GoogleAuthenticatorAuthenticationHandler]: [Secret cannot be null.]>
```

If we can mock this whole thing, and have a user with an active TGT, this is not needed.